docs/examples/multi-agent-workflow.md

```diff
@@ -0,0 +1,233 @@
+            ),
+        ],
+    )
+
+    return researcher
+
+
+def build_writer(role: str, goal: str, backstory: str):
+    """Builds a Writer Structure.
+
+    Args:
+        role: The role of the writer.
+        goal: The goal of the writer.
+        backstory: The backstory of the writer.
+    """
+    writer = Agent(
+        id=role.lower().replace(" ", "_"),
+        event_listeners=[
+            EventListener(
+                event_types=[FinishStructureRunEvent],
+                driver=WebhookEventListenerDriver(
+                    webhook_url=os.environ["ZAPIER_WEBHOOK_URL"],
+                ),
+            )
+        ],
+        rulesets=[
+            Ruleset(
+                name="Role",
+                rules=[
+                    Rule(
+                        value=role,
+                    )
+                ],
+            ),
+            Ruleset(
+                name="Goal",
+                rules=[
+                    Rule(
+                        value=goal,
+                    )
+                ],
+            ),
+            Ruleset(
+                name="Backstory",
+                rules=[Rule(value=backstory)],
+            ),
+            Ruleset(
+                name="Expected Output",
+                rules=[
+                    Rule(
+                        value="Full blog post of at least 4 paragraphs",
+                    )
+                ],
+            ),
+        ],
+    )
+
+    return writer
+
+
+if __name__ == "__main__":
+    # load up the list of writers
+    with open(f"{Path(__file__).parent}/writers.json", "r") as f:
+        writers = json.loads(f.read())
+
+    # Build the crew
+    crew = Workflow()
+    research_task = crew.add_task(
+        StructureRunTask(
+            """Conduct a comprehensive analysis of the latest advancements in AI in 2024.
+                Identify key trends, breakthrough technologies, and potential industry impacts.""",
+            id="research",
+            structure_to_run=build_researcher(),
+        ),
+    )
+    end_task = crew.add_task(NoOpTask())
+    crew.insert_tasks(
+        research_task,
+        [
+            StructureRunTask(
+                """Using insights provided, develop an engaging blog
+                post that highlights the most significant AI advancements.
+                Your post should be informative yet accessible, catering to a tech-savvy audience.
+                Make it sound cool, avoid complex words so it doesn't sound like AI.
+
+                Insights:
+                {{ parent_outputs["research"] }}""",
+                structure_to_run=build_writer(
+                    role=writer["role"],
+                    goal=writer["goal"],
+                    backstory=writer["backstory"],
+                ),
+            )
+            for writer in writers
+        ],
+        end_task,
+    )
+
+    crew.run()
+```
+
+Save this JSON data as `writers.json` in the same directory as the script:
```

Comment:
`writer_profiles.json`? Not married to it.

Reply:
Why waste time say lot word when few word do trick?

Reply:
no argument